flink-runtime/src/main/java/org/apache/flink/runtime/io/network/ConnectionID.java

```diff
@@ -75,15 +84,18 @@ public boolean equals(Object other) {
         }
 
         final ConnectionID ra = (ConnectionID) other;
-        if (!ra.getAddress().equals(address) || ra.getConnectionIndex() != connectionIndex) {
-            return false;
-        }
-
-        return true;
+        return ra.getAddress().equals(address)
+                && ra.getConnectionIndex() == connectionIndex
+                && ra.getResourceID().equals(resourceID);
     }
 
     @Override
     public String toString() {
-        return address + " [" + connectionIndex + "]";
+        return address
+                + " ["
+                + connectionIndex
+                + "]"
+                + " resourceID: "
+                + resourceID.getStringWithMetadata();
```

Comment:
Below is how a connection-id is printed.
```
localhost/127.0.0.1:10000 [10] resourceID: producerLocation
```
It would be nice to have a more concise format. E.g.,
```
localhost/127.0.0.1:10000 (producerLocation) [10]
```